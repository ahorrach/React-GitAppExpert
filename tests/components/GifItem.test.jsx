import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components";


describe('first', () => { 
    const title = 'Saitama';
    const url = 'http://one.com/saitama.jpg';

    test('deve de hacer match con el snapshot', () => { 
       const{ container } = render( <GifItem title= {title} url={url} />)
       expect(container).toMatchSnapshot();
     })

     test('url y Alt indicado', () => { 
        render( <GifItem title= {title} url={url} />)    
        screen.debug();
      //  expect( screen.getByRole('img').src).toBe(url);
       // expect( screen.getByRole('img').alt).toBe(title);
       const { src, alt } = screen.getByRole('img');
       expect(src).toBe(url);
       expect(alt).toBe(title);
      })

      test('deve mostrar el titulo', () => { 
        render( <GifItem title= {title} url={url} />)   
        expect( screen.getByText(title)).toBeTruthy();

       })
 })