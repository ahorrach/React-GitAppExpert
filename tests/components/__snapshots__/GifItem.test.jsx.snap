// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`first deve de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="http://one.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
